ext/sqlite3/sqlite3.c

```diff
@@ -2194,6 +2194,37 @@ static void php_sqlite3_object_free_storage(zend_object *object) /* {{{ */
 }
 /* }}} */
 
+static HashTable *php_sqlite3_get_gc(zend_object *object, zval **table, int *n)
+{
+	php_sqlite3_db_object *intern = php_sqlite3_db_from_obj(object);
+
+	if (intern->funcs == NULL && intern->collations == NULL) {
+		/* Fast path without allocations */
+		*table = NULL;
+		*n = 0;
+	} else {
+		zend_get_gc_buffer *gc_buffer = zend_get_gc_buffer_create();
+
+		php_sqlite3_func *func = intern->funcs;
+		while (func != NULL) {
+			zend_get_gc_buffer_add_zval(gc_buffer, &func->func);
+			zend_get_gc_buffer_add_zval(gc_buffer, &func->step);
+			zend_get_gc_buffer_add_zval(gc_buffer, &func->fini);
+			func = func->next;
+		}
+
+		php_sqlite3_collation *collation = intern->collations;
+		while (collation != NULL) {
+			zend_get_gc_buffer_add_zval(gc_buffer, &collation->cmp_func);
+			collation = collation->next;
+		}
+
+		zend_get_gc_buffer_use(gc_buffer, table, n);
+	}
+
+	return zend_std_get_properties(object);
```

Comment:
Nit: zend_std_get_properties will build the property hashtable if it wasn't already. You can avoid it in two cases:
- In the `if (intern->funcs == NULL && intern->collations == NULL) {` branch, return `zend_std_get_gc(object, table, n)`
- In the other branch, you can skip zend_std_get_properties() if ce->default_properties_count == 0 and object->properties is null, because it means that there are no declared nor dynamic properties

ce->default_properties_count will be > 0 in sub-classes with declared properties, and intern->std.properties will be non-null if dynamic properties was asasigned to the object.